pennylane/compiler/python_compiler/dialects/stablehlo.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Extended StableHLO dialect that dynamically includes all upstream operations
+plus custom operations for PennyLane's compiler infrastructure.
+
+This module automatically imports all operations and attributes from the upstream
+xdsl.dialects.stablehlo and adds custom ones without needing to hardcode
+the upstream operation list.
+"""
+
+import xdsl.dialects.stablehlo as xstablehlo
+from xdsl.dialects.stablehlo import *  # pylint: disable=wildcard-import,unused-wildcard-import
+from xdsl.ir import Dialect
+
+# Operations to add to the dialect
+OPERATIONS = []
```

Comment:
Should there also be a list for ops that already exist upstream but we need to update? For example, `ConstantOp` needs to declare `value` as a property instead of an attribute. We can make that changes upstream, or just update it here. Not a blocker, but just wondering.

Reply:
> but would prefer to populate it in a separate PR.

Agreed 👍